source/extensions/filters/http/common/compressor/compressor.cc

```diff
@@ -506,9 +521,12 @@ bool CompressorFilterConfig::DirectionConfig::isMinimumContentLength(
     }
     return is_minimum_content_length;
   }
-
-  return StringUtil::caseFindToken(headers.getTransferEncodingValue(), ",",
-                                   Http::Headers::get().TransferEncodingValues.Chunked);
```

Comment:
nit, feel free to ignore: I don't have strong opinion, but it wouldn't harm to just move this check above the C-L check as it was in the previous PR. T-E is not removed from response headers by the time this function is called. My only concern is that disabling this feature flag for whatever reason doesn't bring us to the old behavior.

Reply:
it was removed because filter does not see this header (T-E:chunked). there is above discussion w/ @alyssawilk  about it and that i've tried to run a test w/ ASSERT in this code path (basically if (StringUtil::case...){ ASSERT(1 == 2)}) generated a reply w/ T-E:chunked and it was:

Reply:
In the discussion @alyssawilk confirmed that the filter should see the header actually and in my tests the filter sees the header very well. Usually to play with response headers I use this fake HTTP server:
```python
#!/usr/bin/env python3
"""
Very simple HTTP server in python for logging requests
Usage::
    ./server.py [<port>]
"""
from http.server import BaseHTTPRequestHandler, HTTPServer
import logging
import gzip
import shutil


class S(BaseHTTPRequestHandler):

  def _set_response(self):
    self.send_response(200)
    self.send_header('Content-Type', 'text/html')
    self.end_headers()

  def _get_chunk_size(self):
      size_str = self.rfile.read(2)
      while size_str[-2:] != b"\r\n":
          size_str += self.rfile.read(1)
      return int(size_str[:-2], 16)

  def _get_chunk_data(self, chunk_size):
      data = self.rfile.read(chunk_size)
      self.rfile.read(2) # read out '\r\n'
      return data

  def do_GET(self):
    logging.info("GET request,\nPath: %s\nHeaders:\n%s\n", str(self.path), str(self.headers))
    msg = "GET request for {}\nLorem ipsum is a pseudo-Latin text used in web design, typography,\n layout, and printing in place of English\n to emphasise design elements over content.\n It's also called placeholder (or filler)\n text. It's a convenient tool for mock-ups\n".format(
        self.path).encode('utf-8')
    self.send_response(200)
    self.send_header('Content-Type', 'text/plain')
    #self.send_header('Content-Length', str(len(msg)))
    self.send_header('Transfer-Encoding', 'chunked')
    #self.send_header('vary', 'Accept-Encoding')
    self.end_headers()
    self.wfile.write(msg)

  def do_POST(self):
      logging.info("POST request,\nPath: %s\nHeaders:\n%s", str(self.path), str(self.headers))
      if self.headers['Content-Length'] is not None and \
         self.headers['Content-Encoding'] is None:
          content_length = int(self.headers['Content-Length'])  # <--- Gets the size of data
          post_data = self.rfile.read(content_length)  # <--- Gets the data itself
          logging.info("Body:\n%s\n", post_data.decode('utf-8'))

      if self.headers['Content-Encoding'] == 'gzip':
          data = b""
          while True:
              chunk_size = self._get_chunk_size()
              if chunk_size == 0:
                  break
              else:
                  data += self._get_chunk_data(chunk_size)
          logging.info("Uncompressed Body:\n%s\n", gzip.decompress(data).decode('utf-8'))

      self.send_response(200)
      self.send_header('Content-Type', 'text/html')
      self.end_headers()
      self.wfile.write("POST request for {}\n".format(self.path).encode('utf-8'))


def run(server_class=HTTPServer, handler_class=S, port=8080):
  logging.basicConfig(level=logging.INFO)
  server_address = ('', port)
  httpd = server_class(server_address, handler_class)
  logging.info('Starting httpd...\n')
  try:
    httpd.serve_forever()
  except KeyboardInterrupt:
    pass
  httpd.server_close()
  logging.info('Stopping httpd...\n')


if __name__ == '__main__':
  from sys import argv

  if len(argv) == 2:
    run(port=int(argv[1]))
  else:
    run()

```

Reply:
for now i will just move back all TE related logic. it was not part of original diff anyway, so if needed - this could be handled in separate, TE related diff.